qutebrowser/misc/elf.py

```diff
@@ -316,21 +316,13 @@ def parse_webenginecore() -> Optional[Versions]:
     else:
         library_path = pathlib.Path(QLibraryInfo.location(QLibraryInfo.LibrariesPath))
 
-    filenames = [
-        # PyQt bundles those files with a .5 suffix, and Linux distributions
-        # most likely symlinks that.
-        'libQt5WebEngineCore.so.5',
-        # OpenBSD
-        'libQt5WebEngineCore.so.1.0',
-    ]
-    for filename in filenames:
-        lib_file = library_path / filename
-        if lib_file.exists():
-            log.misc.debug(f"QtWebEngine .so found at {lib_file}")
-            break
-    else:
+    library_name = sorted(pathlib.Path(library_path).glob('libQt5WebEngineCore.so*'))
+    if not library_name:
         log.misc.debug(f"No QtWebEngine .so found in {library_path}")
         return None
+    else:
+        lib_file = library_path / library_name[-1]
```

Comment:
Globbing with an absolute path already gives you absolute paths:

```pycon
>>> sorted(pathlib.Path('/usr/lib').glob('libQt5WebEngineCore.so*'))
[PosixPath('/usr/lib/libQt5WebEngineCore.so'), PosixPath('/usr/lib/libQt5WebEngineCore.so.5'), PosixPath('/usr/lib/libQt5WebEngineCore.so.5.15'), PosixPath('/usr/lib/libQt5WebEngineCore.so.5.15.3')]
```

so I don't see the point of joining the resulting paths with `library_path` again.